### PR TITLE
CameraSettingsDialog

### PR DESCRIPTION
**Pull request by PingHsunTsai**
### What type of change is this?

- [ ] `CameraSettingsDialog`component at `toolbar` which handle camera setting popup window.

Screenshot from 2024-05-15 15-56-22

### CODE REVIEW

**Comment by PingHsunTsai** [comment]
`FOV`, `NEAR` and `FAR` method will require a bit of restructure of `camera` which I will do in another pull request.
Feels like the goal will be more clear by starting with all this necessary function before restructure `View3D`, since it takes a bit more time

File: src/compas_viewer/components/dialog.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        Stores references to spin box widgets for easy access when updating camera settings.
+    update_button : QPushButton
+        Button for applying the updated camera settings and closing the dialog.
+
+    Methods
+    -------
+    __init__(self):
+        Initializes the dialog with spin boxes for camera position and target settings.
+    updateCameraTarget(self):
+        Updates the camera's position and target based on the spin box values and closes the dialog.
```
Comment (tomvanmele) [inline src/compas_viewer/components/dialog.py:32]: remove this. replace by a "Parameters" section. see other classes in COMPAS for reference...

File: src/compas_viewer/components/dialog.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    Assuming `viewer` is an instance with a renderer and camera attributes:
+
+    >>> dialog = CameraSettingsDialog()
+    >>> dialog.exec_()  # Executing the dialog to allow user interaction
+
+    References
+    ----------
+    * https://doc.qt.io/qt-5/qdialog.html
+    * https://doc.qt.io/qt-5/qlayout.html
+    * https://doc.qt.io/qt-5/qpushbutton.html
```
Comment (tomvanmele) [inline src/compas_viewer/components/dialog.py:50]: would be better to use qt6 references...

**Review by tomvanmele — APPROVED** [review]
LGTM with a few small things...
much better than the slot in the sidebar btw.
in general, every "function" we add in the toolbar, etc. should also have an entry in the menubar. the menubar should hold "everything". the rest are just shortcuts...

**Comment by tomvanmele** [comment]
ready to merge?

**Comment by PingHsunTsai** [comment]
Yes! if is okay for you

**Pull request merged**